### DISCUSSION TITLE
是否支持加载失败的默认图

### DISCUSSION THREAD

**Discussion by carleft** [Tencent/libpag#1908] [Q&A] [answered]
需求为，从网络上下载pag动画，如果下载失败，用一张静态图代替，如何给PagView或PagImageView设置默认静态图

**Comment by kevingpqi123** [comment, marked as answer]
PAGView 和 PAGImageView 加载的是 pag 文件而不是图片文件，如果需要静态图，可以让设计师设计一个静态效果的 pag 文件，下载失败的时候加载

**Reply by carleft** [reply]
请问是否考虑增加一个展示静态兜底图的功能？
之前尝试过通过bitmap生成图层的方式然后失败了，想请教下这样做为啥不可行
final PAGView pagView= new PAGView(this);
//生成bitmap
Bitmap bitmap = BitmapFactory.decodeResource(getResources(), R.drawable.scene);
int width = bitmap.getWidth();
int height = bitmap.getHeight();
//生成图层
PAGImageLayer imageLayer = PAGImageLayer.Make(width, height, 5000L);
PAGComposition composition = PAGComposition.Make(width, height);
composition.addLayer(imageLayer);
//播放动画
pagView.setComposition(composition);
pagView.play();

**Reply by kevingpqi123** [reply]
暂时没有这样的考虑，我们能够判断的其实外部也可以判断，如文件是否有效等，渲染过程中如果存在异常其实我们内部也没有捕获逻辑
对于 pag 方案而言，已经使用了好多年了，包含微信、手Q 、小红书等国民级应用都在使用，只要能够确保 pag 文件是有效的，渲染绘制层面都不是问题，如果出现，基本上都是必现的，在你们的测试阶段就会暴露出来

至于你上面的代码部分，就创建了一个无内容的 PAGImageLayer，显示效果肯定是空的哈，你前面的 Bitmap 又没有使用到
直接基于 Bitmap 或 路径创建一个 PAGImage，这里面包含有图片的相关信息
然后调用 PAGImageLayer 的 replaceImage 接口，将 PAGImage 传入到 PAGImageLayer，这样 PAGImageLayer 才有数据源

如果不太了解一些 API 接口怎么使用，在 github 仓库中直接搜素，我们公开的 API 方法基本上都有测试用例：
https://github.com/search?q=repo%3ATencent%2Flibpag%20PAGImageLayer%3A%3AMake&type=code

<img width="1246" alt="企业微信截图_7f06aad1-456e-4fe0-916a-922a03a35ea8" src="https://github.com/Tencent/libpag/assets/16609221/e3067102-8543-4b23-a320-10bf4d833b8c">

**Reply by carleft** [reply]
之前漏了生成PAGImage的过程
PAGImage image = PAGImage.FromBitmap(bitmap);
imageLayer.replaceImage(image);
加上之后好了，感谢！
